### ISSUE TITLE
How to follow DOM changes?

### ISSUE THREAD

**Issue by alchen** [vuejs/Discussion#170]
My goal is to keep the scroll position of some `v-repeat` lists. Those list are structured as `component (different lists)` > `component (the list)` > `component (list item)` and they can be either prepended or appended, with the top most component keeping track of the data and passing them down with `v-with`.

The problems is that when things are prepended, old content get pushed down. And I used to do this to keep the scroll position:

1. If appending, record distance to bottom.
2. Add new items to the front.
3. Restore distance to bottom.

But I couldn't find anything to guarantee that `3.` happens after `2.`, except with transitions.

Right now, I'm setting transitions on the list to be either a `noop` or the above procedure (in `beforeEnter` and `enter`), depending on whether the list is getting appended or prepended, which gets communicated with events.

Is there a cleaner/better way to do this? Should things be organized differently?

**Update:** I guess I could I could use the transition to test if the newly added item is at the top and skip the enable/disable transition logic.

Basically, use `keepScroll` transition and do this:

```
    keepScroll: {
      enter: function (el) {
        var container = getContainerElement();
        if (el.offsetTop < threshold // Test for new item position
            // No need to keep scroll position if already close to top
            && container.scrollTop >= threshold) {
          timelineEl.scrollTop += el.scrollHeight;
        }
        done();
      }
    }
```

**Comment by yyx990803** [comment]
In Vue DOM updates follow a simple rule: it happens asynchronously in the next tick after you change the data. So, call `Vue.nextTick(callback)` right after you push something into the list, and in the callback the DOM would have been updated.

**Comment by alchen** [comment]
Thanks! I think that's exactly what I need.
